File: src/p1_two_sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

from typing import List


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        record = {}

        for (i, num) in enumerate(nums):
            matched = record.get(num)
            if matched != None:
                return [matched, i]

            record[target - num] = i


# @lc code=end

if __name__ == "__main__":
    s = Solution()
    print(s.twoSum([2, 7, 11, 15], 9))

File: src/p1_two_sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

package leetcode

// @lc code=start
func twoSum(nums []int, target int) []int {
	record := make(map[int]int)

	for i, num := range nums {
		if matched, ok := record[num]; ok {
			return []int{matched, i}
		}

		record[target-num] = i
	}
	return []int{}
}

// @lc code=end
